### ISSUE TITLE
Using Real sense camera, the CPU usage is reaching 100% in own Plus built

### ISSUE THREAD

**Issue by carlos-luque** [PlusToolkit/PlusLib#643]
Hello everyone, 
  
I am writing if you can give me some hints to fix the CPu usage when the Real sense uses. 
I built The Plus toolkit (master branch) with real sense and thermal expert Q1 cameras without errors, but when real sence is active, the CPU usage is 100%.  I am using the latest sdk real sense version, and the compiler is Visual studio 2015 community. 

I´ve checked the official Plus built when the Real sense is active, the CPU usage is only 25-30 %.  
I would like to know if I should activate any additional variable in compilation time or use a specific real sence sdk version. 

Thanks in advance

**Comment by adamrankin** [comment]
Might be a bit too basic, but did you build in Debug or Release?

**Comment by carlos-luque** [comment]
I built in Release

**Comment by lassoan** [comment]
It is a known behavior of the RealSense SDK that it hogs all the CPU cores. Previously, we had to reduce the thread priority of RealSense to let other threads work.

@Sunderlandkyl do we have any better/smarter mechanism now? Or have they made number of used CPU cores configurable in RealSense SDK?

**Comment by Sunderlandkyl** [comment]
I don't think we've changed the mechanism.
I can take a look and see if the current/latest SDK has any adjustable parameters.

**Comment by carlos-luque** [comment]
I managed to fix the issue.  The issue is the RealSense SDK version.  I found this link https://github.com/IntelRealSense/librealsense/issues/744. The old SDK versions use the OpenMP and it produces the high CPU usage. 

I checked the Real sense SDK version I used to build the Plus and it is 2.11.0. With this version, the CPU utilization is high (100%). I changed the SDK to the version 2.31.0 and  I built the Plus.  With this, The CPU usage is slow (10-20 %).

In summary, I recommend using the latest version of the RealSense SDK.

**Comment by markasselin** [comment]
Hi Carlos,

Thanks for your work looking into this. Did you need to make any changes to the PLUS source code to update the RealSense SDK to 2.31? If not this should be easy to update, just in time for project week!

Cheers, --Mark

**Comment by lassoan** [comment]
Good detective work @carlos-luque. If we find that latency/frame rate drops too much without OpenMP then we may build with options that limit the number of threads that OpenMP can use (OMP_NUM_THREADS) to half of the number of CPU cores.

**Comment by carlos-luque** [comment]
@markasselin No, I did not. I downloaded the sdk binary "Intel.RealSense.SDK-WIN10-2.31.0.1234.exe" and I installed it and then I built the Plus without any issue.  For my current requirements, it works correctly.   

I checked this solution in several computers and the Plus toolkit worked correctly. 

@lassoan, If we want to optimize the sdk, it can be built with other configurations than the normal in sdk binary.

**Comment by adamrankin** [comment]
Closing this in favour of #698
